src/GiraffeGenerator.Core/OpenApi.fs

```diff
@@ -97,59 +162,104 @@ and ObjectKind =
     static member Create(name: string, parameters: OpenApiParameter seq) =
         let fields =
             [ for param in parameters ->
-                param.Name, TypeKind.Parse param.Schema ]
+                let kind, def = TypeKind.Parse param.Schema
+                let kind = ObjectKind.OptionIfNotRequired (param.Required || def.IsSome) kind
+                param.Name, kind, def ]
             
         { Name = Some name
           Properties = fields
           Docs = None }
-
+and DUCaseKind =
+    {
+        Docs: Docs option
+        CaseName: string option
+        Kind: TypeKind
+    }
+and DUKind =
+    {
+        Name: string
+        Cases: DUCaseKind list
+        Docs: Docs option
+    }
 /// Kind of types for fields in schemas
-/// Either primitive or Array<T> or Option<T> or Object with properties
+/// Either primitive or Array<T> or Option<T> or Object with properties or DU
 /// Array, Option and Object could recursively contains similar types
 and TypeKind =
     | Prim of PrimTypeKind
-    | Array of TypeKind
+    | Array of TypeKind * Option<DefaultableKind>
```

Comment:
Defaults could be only for primitives right? Could be just move it to there?
Not sure I got idea of having Arrays with default values correctly

Reply:
I can't find any restriction on default applicable only to primitive types in OpenAPI spec. [This article](https://swagger.io/blog/api-development/unlocking-the-spec-the-default-keyword/) mentions that complex types and arrays may have a default value. That's why I've decided to include default values for them too - support for primitive types only seems more like an implementation detail. Of course, I could promote this implementation detail to the specification model, which would make the code slightly simpler for now but would also make an introduction of support for default values in complex and array types slightly more difficult. If you see this as a better way, I'll do it.

Reply:
I think it's fine as is for now. If there will be design issues, we could refactor later